core/domain/summary_services.py

```diff
@@ -131,3 +140,23 @@ def _generate_query(categories):
         })
 
     return results
+
+def get_featured_explorations():
+    """Returns a list of featured explorations."""
+    exp_summary_dict = []
+    exp_summary_dict = exp_services.get_non_private_exploration_summaries()
```

Comment:
This function actually needs to be rewritten...

When I suggested following get_non_private_exploration_summaries(), I meant following the path it takes through the backend, not using it directly. The current implementation runs into the same issue I mentioned before, namely, you're pulling out a large bunch of explorations and then returning a small subset of them. This is pretty wasteful.

Instead, it would be better to go all the way to the storage layer and execute a query on the models to see which ones have publicized status. Have a look at how exp_models.ExpSummaryModel.get_non_private() is referenced in get_non_private_exploration_summaries() -- you'll want to follow a similar pattern, and add something like get_featured() to ExpSummaryModel.

Note that some of the comments below are now irrelevant, but I've left them in just FYI.